client/src/components/HistoryViewer/HistoryViewer.js

```diff
@@ -33,6 +39,30 @@ class HistoryViewer extends Component {
     return edges.map((version) => version.node);
   }
 
+  /**
+   * Renders the detail form for a selected version
+   *
+   * @returns {HistoryViewerVersionDetail}
+   */
+  getVersionDetail() {
+    const { currentVersion, recordId, recordClass, handleSetCurrentVersion, store } = this.props;
+
+    const sectionConfigKey = 'SilverStripe\\VersionedAdmin\\Controllers\\HistoryViewerController';
+    const sectionConfig = store.getState().config.sections
+      .find((section) => section.name === sectionConfigKey);
```

Comment:
Config properties should be injected via HOC. In asset-admin, for example, the boot code uses this.

```js
const sectionConfig = Config.getSection('SilverStripe\\AssetAdmin\\Controller\\AssetAdmin');

reactRouteRegister.add({
    path: sectionConfig.url,
// ...
```

Maybe injecting schemaUrl could be done by the HistoryViewerRouter HOC?

HistoryViewer.js should have a schemaURL that's injected by the HOC.

Then in your unit tests, you can just inject a hard-coded URL via props.

If you like, schemaURL could have wildcards. E.g. `:id` replaced with the recordID, `:recordClass` the record class, and so on.

Instead of extracting this from redux use `import Config from 'lib/Config';` instead. :D Much less complex.

Reply:
If you do this, you can completely omit the `window.ss.store` from the HOC anyway. :)

Reply:
We shouldn't be passing the store as a prop. Instead, use the `connect` HOC to provide the slice of config you need from the store as a prop, as @tractorcow has articulated.

The only place `store` should be passed as a prop is on the top-level component that creates the redux context. It can be accessed directly (e.g. `getState()`) but only in action creators.

Reply:
Ok I've updated this to have a HistoryViewerConfig HoC which gets the store config and provides it as a prop to HistoryViewer, and the `schemaUrl` is now an injected prop too with placeholders for the variables.